File: DigitalLicenseApplication/Frontend/src/app/services/department/departmentservice.ts
import { Injectable } from '@angular/core';
import { Department, DepartmentPostDTO } from '../../models/department.model';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root',
})
export class Departmentservice {
  private apiUrl = 'https://localhost:7034/api/Department';

  constructor(private http: HttpClient) {}

  getAllDepartments(): Observable<Department[]> {
    return this.http.get<Department[]>(this.apiUrl);
  }

  getDepartmentById(id: number): Observable<Department> {
    return this.http.get<Department>(`${this.apiUrl}/${id}`);
  }

  addDepartment(dto: DepartmentPostDTO): Observable<Department> {
    return this.http.post<Department>(this.apiUrl, dto);
  }

  updateDepartment(department: Department): Observable<void> {
    return this.http.put<void>(
      `${this.apiUrl}/${department.departmentId}`,
      department
    );
  }

  deleteDepartment(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: DigitalLicenseApplication/Backend/DigitalLicenseApplication/DigitalLicenseApplication/Service/Departmentservice.cs
using DigitalLicenseApplication.Interface.Repository;
using DigitalLicenseApplication.Interface.Service;
using DigitalLicenseApplication.Models;
using System.Collections.Generic;

namespace DigitalLicenseApplication.Service
{
    public class Departmentservice : IDepartmentservice
    {
        private readonly IDepartmentrepo _departmentrepo;

        public Departmentservice(IDepartmentrepo departmentrepo)
        {
            _departmentrepo = departmentrepo;
        }

        public IEnumerable<Department> GetAllDepartments()
        {
            return _departmentrepo.GetAllDepartments();
        }

        public Department GetDepartmentById(int id)
        {
            return _departmentrepo.GetDepartmentById(id);
        }

        public void AddDepartment(Department department)
        {
            _departmentrepo.AddDepartment(department);
        }

        public void UpdateDepartment(Department department)
        {
            _departmentrepo.UpdateDepartment(department);
        }

        public void DeleteDepartment(int id)
        {
            _departmentrepo.DeleteDepartment(id);
        }
    }
}
